### PR TITLE
Oneliner/basic implementation tests

### PR DESCRIPTION
**Pull request by nstjean**
Adding tests for basic implementation.

### CODE REVIEW

**Comment by nstjean** [comment]
Basic (One-liner) Implementation Tests are now working.

I still have to work on the PLpeople tests and possibly clean up some others.

**Comment by nstjean** [comment]
The only way I can get the LEL tests working is by running the entire `dist/LeafletEnvironmentalLayers.js` through babel. That might be what is causing the build error, if it's really slow.

**Comment by nstjean** [comment]
I've gone back to the original Gruntfile and the build is still failing.

**Comment by nstjean** [comment]
Yes, it's this that is causing the issue, under `babel: dist: files:`
`"dist/LeafletEnvironmentalLayers_babel.js": "dist/LeafletEnvironmentalLayers.js",`
I think it's timing out while trying to process it.

Ideas? Should we increase the wait time?

**Comment by nstjean** [comment]
Basically it looks like no matter how I try I can't run Leaflet through babel without an error. But I need a transpiled Leaflet to be able to use LEL in the tests.

**Comment by nstjean** [comment]
`The job exceeded the maximum log length, and has been terminated.`
So maybe it's not a time issue but just an output/log size issue. I'll look into ways to reducing the output that transpile produces.

File: Gruntfile.js [hunk 1]
```diff
@@ -12,16 +12,16 @@ module.exports = function(grunt) {
...
-            dist: {
-                src: ['node_modules/jquery/dist/jquery.min.js', 'node_modules/leaflet/dist/leaflet.js', 'src/leafletEnvironmentalLayers.js', 'src/util/*.js'],
-                dest: 'dist/LeafletEnvironmentalLayers.js'
-            },
-            babel: {
-              files: {
-                "dist/util/layersBrowser.js": "dist/util/layersBrowser_babel.js"
-              }
+          dist: {
+              src: ['node_modules/leaflet/dist/leaflet.js', 'src/leafletEnvironmentalLayers.js', 'src/util/*.js'],
```
Comment (nstjean) [inline Gruntfile.js]: @crisner Do you know if we have to have jquery in the dist file? I removed it to make the file size smaller and everything seems to work fine, but I want to make sure there's not some other reason it's in there.

File: Gruntfile.js [hunk 2]
```diff
@@ -33,7 +33,7 @@ module.exports = function(grunt) {
 
         babel: {
           options: {
-            sourceMap: true,
+            sourceMaps: false,
```
Comment (nstjean) [inline Gruntfile.js]: @crisner Also the sourceMaps - do we need to have the .map files available? The tests don't require them. @sagarpreet-chadha

File: Gruntfile.js (hunk 1 above)
Comment (crisner) [reply Gruntfile.js]: Not sure about that myself. @sagarpreet-chadha, do you know if it is required to include jquery here?

File: Gruntfile.js (hunk 2 above)
Comment (crisner) [reply Gruntfile.js]: From what I understand sourceMaps help in debugging, provides line numbers to narrow down where the issue is from.

Comment (nstjean) [reply Gruntfile.js]: Ok, I can add that back in easy enough.

File: Gruntfile.js [hunk 3]
```diff
@@ -35,9 +35,10 @@ module.exports = function(grunt) {
           options: {
             sourceMaps: false,
             presets: ['@babel/preset-env'],
-            "plugins": [
-              "transform-object-rest-spread",
-              "transform-remove-strict-mode"
+            plugins: [
+              // "transform-remove-console",
+              // "transform-object-rest-spread",
```
Comment (crisner) [inline Gruntfile.js]: Are the tests working when this is removed? There was a file that used spread operators and some browsers throw an error if it doesn't have support. I had the issue, I wonder if it works now?

**Comment by nstjean** [comment]
Got it to work! I removed `babel-plugin-transform-object-rest-spread` - it doesn't exist on github anymore, and the transpile and tests all work without it. I'm guessing the core of babel handles the spread operation now.

This is comparatively slow to compile, but this way instead of worrying about which individual files need to be run through babel we can just do it all at once in `dist/LeafletEnvironmentalLayers_babel.js`, including Leaflet.

@crisner It looks like `src/util/layersBrowser.js` goes through babel twice - once under `babel : dist` and then again under `browserify : babel`. The strange part is that if I remove one of them then I get an error in the tests.
`ReferenceError: Can't find variable: require in file:///home/natalie/Dev/Ruby/public-lab/leaflet-environmental-layers/dist/util/layersBrowser_babel.js (line 521)`

**Comment by jywarren** [comment]
Oh, this is great. It's also good i think to run babel in Travis, because it adds an extra layer of confirmation that new dependencies and such don't break the build. Thanks!

**Comment by crisner** [comment]
> It looks like src/util/layersBrowser.js goes through babel twice - once under babel : dist and then again under browserify : babel. The strange part is that if I remove one of them then I get an error in the tests.

The first one is babelifying the file. The second one is browserifying it. Something to do with allowing the file to be "required" in the browser.

**Comment by nstjean** [comment]
> > It looks like src/util/layersBrowser.js goes through babel twice - once under babel : dist and then again under browserify : babel. The strange part is that if I remove one of them then I get an error in the tests.
> 
> The first one is babelifying the file. The second one is browserifying it. Something to do with allowing the file/variables in the file to be "required" in the browser.

Ahhh, ok!

**Comment by nstjean** [comment]
> Oh, this is great. It's also good i think to run babel in Travis, because it adds an extra layer of confirmation that new dependencies and such don't break the build. Thanks!

Yes, for sure. And it needs to run babel in order to run the jasmine tests!

**Comment by nstjean** [comment]
This is all complete and runs! I think it's ready for review! @crisner @sagarpreet-chadha

**Comment by crisner** [comment]
@nstjean, what are the files in 'text-table' for?

**Comment by nstjean** [comment]
> @nstjean, what are the files in 'text-table' for?

That's weird... I have no idea what those are. Let me check into where they are coming from.

File: dist/LeafletEnvironmentalLayers.css [hunk 4]
```diff
@@ -1,418 +0,0 @@
-.owm-icondiv {
```
Comment (crisner) [inline dist/LeafletEnvironmentalLayers.css]: Was this file deleted accidentally? This is the main stylesheet for LEL.

**Comment by crisner** [comment]
This is how it looks now:
image
image

This is how they'll look with styles applied:
image
image

**Comment by crisner** [comment]
I checked out the tests locally. They are working perfectly! 😄 
And thanks for cleaning up the example files! 🎉

**Comment by nstjean** [comment]
Must have been deleted accidentally! Added it back in

**Comment by cesswairimu** [comment]
Wow this is great :tada: :tada: We ready to merge?

**Comment by nstjean** [comment]
Yes! :)

**Comment by cesswairimu** [comment]
@nstjean could you please fix the conflicting files

**Comment by cesswairimu** [comment]
Rad :tada: :tada:

**Pull request merged**